xarray/core/parallel.py

```diff
@@ -310,9 +313,13 @@ def _wrapper(func, obj, to_array, args, kwargs):
                     # unchunked dimensions in the input have one chunk in the result
                     key += (0,)
 
-            graph[key] = (operator.getitem, from_wrapper, name)
+            new_layers[gname_l][key] = (operator.getitem, from_wrapper, name)
 
-    graph = HighLevelGraph.from_collections(gname, graph, dependencies=[dataset])
+    hlg = HighLevelGraph.from_collections(gname, graph, dependencies=[dataset])
```

Comment:
Changed to `hlg` primarily to please mypy, since we declare `graph` to be a Dict.